### PR TITLE
feat: possibility to sum two portfolios

### LINKED ISSUE #113: ODHack: Make it easier to sum 2 Portfolio instances
Adjust the Portfolio class so that two of its instances can be summed together, resulting in a portfolio containing tokens from both summed portfolios. The Portfolio class is implemented here: [https://github.com/CarmineOptions/derisk-research/blob/master/src/helpers.py#L78](https://github.com/CarmineOptions/derisk-research/blob/master/src/helpers.py#L78).

### PR DESCRIPTION
**Pull request by lindsaymoralesb**
# Issue:

Resolves #113 

# Summary:

Modified the portfolio class to add the possibility of adding two instances together.
For this I first added the validation of the second instance being of the same type (portfolio), if not, raise an error.
Then take the tokens values and add them together to return a new portfolio instance containing these new values.

### CODE REVIEW

File: src/helpers.py [hunk 1]
```diff
@@ -83,6 +83,17 @@ class Portfolio(TokenValues):
...
+    def __add__(self, second_portfolio):
+        if not isinstance(second_portfolio, Portfolio):
+            raise TypeError(f"Cannot add Portfolio with {type(second_portfolio)}")
+        added_tokens_values = self.values.copy()
+        for token, amount in second_portfolio.values.items():
+            if token in added_tokens_values:
+                added_tokens_values[token] += amount
+            else:
+                added_tokens_values[token] = amount
+        return Portfolio(added_tokens_values)
```
Comment (lukaspetrasek) [inline src/helpers.py:95]: I'm getting `TypeError: __init__() takes 1 positional argument but 2 were given` on this line. The reason is that `Portfolio` can't currently be initialized with a dict of tokens and amounts. Pls add this functionality.

Comment (lindsaymoralesb) [reply src/helpers.py:98]: My bad, done!

File: src/helpers.py [hunk 2]
```diff
@@ -80,22 +80,22 @@ class Portfolio(TokenValues):
 
     MAX_ROUNDING_ERRORS: TokenValues = MAX_ROUNDING_ERRORS
 
-    def __init__(self, token_values=None) -> None:
+    def __init__(self, token_values: Optional[dict[str, Union[bool, decimal.Decimal]]] = None) -> None:
```
Comment (lukaspetrasek) [inline src/helpers.py:83]:
```suggestion
    def __init__(self, token_values: Optional[dict[str, bool | decimal.Decimal]] = None) -> None:
```

File: src/helpers.py [hunk 3]
```diff
@@ -80,22 +80,22 @@ class Portfolio(TokenValues):
...
         if token_values is None:
             super().__init__(init_value=decimal.Decimal("0"))
         else:
             super().__init__(values=token_values)
 
-    def __add__(self, second_portfolio):
+    def __add__(self, second_portfolio: 'Portfolio') -> 'Portfolio':
```
Comment (lukaspetrasek) [inline src/helpers.py:89]:
```suggestion
    def __add__(self, second_portfolio: Portfolio) -> Portfolio:
```

**Review by lukaspetrasek — CHANGES_REQUESTED** [review]
Almost there!

**Comment by lukaspetrasek** [comment]
Wonderful! Merging.

**Pull request merged**